File: benchmarks/evidence_reuse/fixtures/c_fixture/tests/test_app.c
#include "app.h"

#include <math.h>
#include <stdio.h>
#include <string.h>


#define CHECK(expression) do { \
    if (!(expression)) { \
        fprintf(stderr, "check failed: %s\n", #expression); \
        return 1; \
    } \
} while (0)


int main(void) {
    CHECK(fabs(total_with_tax(10.0) - 11.5) < 0.000001);
    CHECK(strcmp(greeting("  Ada "), "Hello, Ada!") == 0);
    CHECK(strcmp(fixture_label(), "control") == 0);
    CHECK(strcmp(mode_label(), "test") == 0);
    return 0;
}

File: benchmarks/evidence_reuse/fixtures/python_fixture/tests/test_app.py
import unittest

from app import fixture_label, greeting, mode_label, total_with_tax


class AppTests(unittest.TestCase):
    def test_total_with_tax(self) -> None:
        self.assertEqual(total_with_tax(10), 11.5)

    def test_greeting(self) -> None:
        self.assertEqual(greeting("  Ada "), "Hello, Ada!")

    def test_fixture_label(self) -> None:
        self.assertEqual(fixture_label(), "control")

    def test_mode_label(self) -> None:
        self.assertEqual(mode_label(), "test")


if __name__ == "__main__":
    unittest.main()
